types/block_test.go

```diff
@@ -268,6 +268,52 @@ func TestCommitValidateBasic(t *testing.T) {
 	}
 }
 
+func TestMaxCommitSigBytes(t *testing.T) {
+	// time is varint encoded so need to pick the max.
+	// year int, month Month, day, hour, min, sec, nsec int, loc *Location
+	timestamp := time.Date(math.MaxInt64, 0, 0, 0, 0, 0, math.MaxInt64, time.UTC)
+
+	cs := &CommitSig{
+		BlockIDFlag:      BlockIDFlagNil,
+		ValidatorAddress: crypto.AddressHash([]byte("validator_address")),
+		Timestamp:        timestamp,
+		Signature:        tmhash.Sum([]byte("signature")),
+	}
+
+	pb := cs.ToProto()
+
+	assert.EqualValues(t, MaxCommitSigBytes, pb.Size())
+}
+
+func TestMaxCommitBytes(t *testing.T) {
+	timestamp := time.Date(math.MaxInt64, 0, 0, 0, 0, 0, math.MaxInt64, time.UTC)
+
+	cs := CommitSig{
+		BlockIDFlag:      BlockIDFlagNil,
+		ValidatorAddress: crypto.AddressHash([]byte("validator_address")),
+		Timestamp:        timestamp,
+		Signature:        tmhash.Sum([]byte("signature")),
+	}
+
+	commit := &Commit{
+		Height: math.MaxInt64,
+		Round:  math.MaxInt32,
+		BlockID: BlockID{
+			Hash: tmhash.Sum([]byte("blockID_hash")),
+			PartSetHeader: PartSetHeader{
+				Total: math.MaxInt32,
+				Hash:  tmhash.Sum([]byte("blockID_part_set_header_hash")),
+			},
+		},
+		Signatures: []CommitSig{cs},
```

Comment:
You may have to calculate this on the fly because currently both `Commit` and `CommitSig` doesn't account for repeated field size. (https://github.com/tendermint/tendermint/pull/5483#issuecomment-707612359)

Reply:
Is this constant size?

Reply:
No, I think this is right. There will be 1 repeated value for the Signatures field here which will include the key overhead, and the key (and therefore the overhead) is the same for each repeat so it grows linearly.

Reply:
> Is this constant size?

Lol...
```
Error Trace:	block_test.go:322
        	Error:      	Not equal: 
        	            	expected: 7796
        	            	actual  : 7994
        	Test:       	TestMaxCommitBytes
```

Reply:
Apparently it's 2 bytes extra per commit sig

Reply:
I think this should be good now

Reply:
Yeah, the field key is 2 bytes unless we change it to something that exceeds a two-byte varint - in which case this test will warn us about it by failing anyway.